### ISSUE TITLE
Failed to link.Object reference

### ISSUE THREAD

**Issue by Darkstars123** [Kwoth/NadekoBot#577]
Recently i've been having an issue where music commands seem to just make Nadeko join the call and sit there. She doesn't send any messages in chat and sends an error in the console (http://prnt.sc/c6lu0p) Any idea what this means?

**Comment by Nitix** [comment]
Seems like your youtube query has failed, make sure you enabled youtube for your google api key. It's not enabled by default.

**Comment by Darkstars123** [comment]
It should be enabled (http://prntscr.com/c6m2aj)

**Comment by appelemac** [comment]
This seems to happen for me for some queries that don't seem...specific enough. ie. `!!q hello` doesn't work, but `!!q adele hello`does

**Comment by justinkruit** [comment]
It happens here too. Reinstalled the bot to be sure, and it still didn't fix it. Some stuff changing on YouTube's side?

**Comment by miraai** [comment]
Have in mind that yt api cant guess some searches, and wont play it.

**Comment by justinkruit** [comment]
@miraai ended up testing it with a link of a song multiple times, didn't work either

**Comment by miraai** [comment]
What version are you on?

**Comment by justinkruit** [comment]
I'm on v0.99.9x. But I just looked in my terminal again, and now I get this:
`Severity: ErrorMessage: DisconnectedExceptionMessage: You must send data by websocket after websocket is opened!`
And that is all the information I get. It happens whenever I queue a new song. It shows the song in the chat, but it won't join the voice channel even anymore now.

**Comment by miraai** [comment]
Did you check bot permissions? When you open cmd, and type `ffmpeg` do you get that you have it installed?

**Comment by justinkruit** [comment]
The bot has all the permissions, and I'm pretty sure ffmpeg is installed, because it worked before.

**Comment by miraai** [comment]
Can you check it like i told you and show me the screenshot?

**Comment by Genskoria** [comment]
I'm having the same issue with the Music commands, sometimes will work... but in a lot of cases the result is not  like that.
I did the cmd/ffmpeg thing after a reinstall of the same and it showed me this: http://prntscr.com/c7y3lq
Also some sc from the problem: http://prntscr.com/c7y4vp
And Info of the build i'm using i think: http://prntscr.com/c7y5ov

**Comment by Nitix** [comment]
I found the problem using your query, Google may return either a playlist or a video. But Nadeko doesn't know what to do with the playlist and fail to resolve the query.

Exemple of expected response from Youtube API :
```
{
 "kind": "youtube#searchListResponse",
 "etag": "\"I_8xdZu766_FSaexEaDXTIfEWc0/yLN0Ms_z_cfO4aUlfMTIs4hMNv8\"",
 "nextPageToken": "CAEQAA",
 "regionCode": "FR",
 "pageInfo": {
  "totalResults": 1000000,
  "resultsPerPage": 1
 },
 "items": [
  {
   "kind": "youtube#searchResult",
   "etag": "\"I_8xdZu766_FSaexEaDXTIfEWc0/j0uEstXCXOhrDqDegEBmEeHqsBM\"",
   "id": {
    "kind": "youtube#video",
    "videoId": "YQHsXMglC9A"
   },
   "snippet": {
    "publishedAt": "2015-10-23T06:54:18.000Z",
    "channelId": "UComP_epzeKzvBX156r6pm1Q",
    "title": "Adele - Hello",
    "description": "'Hello' is taken from the new album, 25, out November 20. http://adele.com Available now from iTunes http://smarturl.it/itunes25 Available now from Amazon ...",
    "thumbnails": {
     "default": {
      "url": "https://i.ytimg.com/vi/YQHsXMglC9A/default.jpg",
      "width": 120,
      "height": 90
     },
     "medium": {
      "url": "https://i.ytimg.com/vi/YQHsXMglC9A/mqdefault.jpg",
      "width": 320,
      "height": 180
     },
     "high": {
      "url": "https://i.ytimg.com/vi/YQHsXMglC9A/hqdefault.jpg",
      "width": 480,
      "height": 360
     }
    },
    "channelTitle": "AdeleVEVO",
    "liveBroadcastContent": "none"
   }
  }
 ]
}
```
Unexpected response from Youtube API (playlist)
```
{
 "kind": "youtube#searchListResponse",
 "etag": "\"I_8xdZu766_FSaexEaDXTIfEWc0/RLbgsvp0DEn90HNGFt4zXxPfwX0\"",
 "nextPageToken": "CAEQAA",
 "regionCode": "FR",
 "pageInfo": {
  "totalResults": 14631,
  "resultsPerPage": 1
 },
 "items": [
  {
   "kind": "youtube#searchResult",
   "etag": "\"I_8xdZu766_FSaexEaDXTIfEWc0/Nzf5tXFDihJhXsdcg3q3-WoiyZ8\"",
   "id": {
    "kind": "youtube#playlist",
    "playlistId": "PL48602B13CF855D13"
   },
   "snippet": {
    "publishedAt": "2010-02-19T19:37:30.000Z",
    "channelId": "UCTvbAOu6pTvS1Xr2rVXiFWw",
    "title": "ICC soundtrack",
    "description": "",
    "thumbnails": {
     "default": {
      "url": "https://i.ytimg.com/vi/iiuxGeYj_2E/default.jpg",
      "width": 120,
      "height": 90
     },
     "medium": {
      "url": "https://i.ytimg.com/vi/iiuxGeYj_2E/mqdefault.jpg",
      "width": 320,
      "height": 180
     },
     "high": {
      "url": "https://i.ytimg.com/vi/iiuxGeYj_2E/hqdefault.jpg",
      "width": 480,
      "height": 360
     }
    },
    "channelTitle": "sarmay",
    "liveBroadcastContent": "none"
   }
  }
 ]
}
```
Anyway, currently @Kwoth and others and converting Nadeko with .Net Core and other with the 1.0. I'm pretty sure, It would resolve the issue with the release of 1.0. So wait a bit :). 

While waiting, if Nadeko failed to resolve try to use `!!pl query` as a workaround or directly link the video.

Thanks for posting the current exemple of failing queries, now the bug may be more 'easier' to fix.

**Comment by Genskoria** [comment]
Thanks for answering so quickly @Nitix , i'll be looking forward to the next update. :smile:

**Comment by Kwoth** [comment]
@Nitix That is correct. I figured out the same thing while actually looking into google api library, which we will be using - i didn't set the type to video. 

It will indeed be corrected in 1.0

PS *I'm very surprised you figured it out, good job*

**Comment by Darkstars123** [comment]
Glad to see it will be fixed! Best bot is getting better all the time c:
